src/UnifiedFFI/FFIReferenceValueHolder.class.st

```diff
@@ -28,6 +30,14 @@ FFIReferenceValueHolder >> typeClass [
 	^ self type class
 ]
 
+{ #category : 'private' }
+FFIReferenceValueHolder >> typeSize [
+
+	^ (self typeClass inheritsFrom: FFIStructure)
+		ifTrue: [ self typeClass structureSize ]
+		ifFalse: [ FFIExternalType pointerSize ]
```

Comment:
This is also a very strange point related to the type issue I pointed above.
How come an (untyped) value holder has a pointer to a pointer by default? 

Also, do we really need the `inheritsFrom:`? I mean, if the value holder has a type (structures define types) we can just replace this by

```
self typeClass size
```

And of course a pointer type has the size of a pointer.

Reply:
> Also, do we really need the `inheritsFrom:`? I mean, if the value holder has a type (structures define types) we can just replace this by
> 
> ```
> self typeClass size
> ```
> 
> And of course a pointer type has the size of a pointer.

this is not exactly like that but this  a good observation :)  
I addressed the problem in latest commit.